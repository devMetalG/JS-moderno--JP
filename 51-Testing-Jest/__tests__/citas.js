import Citas from '../js/classes/Citas'

describe('Probar clase Citas',() => {
  
  const citas = new Citas()
  const id = Date.now()
  
  test('agregar nueva cita',()=>{
    const citaObj = {
      mascota: 'Morris',
      id,
      propietario: 'Memo',
      telefono: '1111112222',
      fecha: '11-10-2023',
      hora: '15:10',
      sintomas: 'Esta muy guapo'
    }

    citaObj.id = Date.now()
    
    citas.agregarCita(citaObj)

    expect(citas).toMatchSnapshot()
  })

  test('Actualizar Cita', () => {
    const citaActualizada = {
      mascota: 'Euphy',
      id,
      propietario: 'Costal',
      telefono: '1111112222',
      fecha: '11-10-2023',
      hora: '15:10',
      sintomas: 'sindrome del gato paracaidista'
    }

    citas.editarCita(citaActualizada)

    expect(citas).toMatchSnapshot()
  })
  
  test('Eliminar Cita', () => {

    citas.eliminarCita(id)

    expect(citas).toMatchSnapshot()
  })
})