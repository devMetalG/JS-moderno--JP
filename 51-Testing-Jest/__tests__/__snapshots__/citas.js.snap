// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar clase Citas Actualizar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "11-10-2023",
      "hora": "15:10",
      "id": 1699652123783,
      "mascota": "Morris",
      "propietario": "Memo",
      "sintomas": "Esta muy guapo",
      "telefono": "1111112222",
    },
  ],
}
`;

exports[`Probar clase Citas Eliminar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "11-10-2023",
      "hora": "15:10",
      "id": 1699652123783,
      "mascota": "Morris",
      "propietario": "Memo",
      "sintomas": "Esta muy guapo",
      "telefono": "1111112222",
    },
  ],
}
`;

exports[`Probar clase Citas agregar nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "11-10-2023",
      "hora": "15:10",
      "id": 1699652123783,
      "mascota": "Morris",
      "propietario": "Memo",
      "sintomas": "Esta muy guapo",
      "telefono": "1111112222",
    },
  ],
}
`;
